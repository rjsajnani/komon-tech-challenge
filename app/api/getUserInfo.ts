import axios from 'axios';

const API_BASE = 'https://api.npoint.io/370fc4db2db84ad88075';

export interface LatestPost {
  postId: number;
  postImage: string;
  type: 'img' | 'video';
}

export interface ResponseData {
  avatar_url: string;
  bio: string;
  followers: string;
  id: number;
  name: string;
  lastestPost: LatestPost[];
}

export const getUserInfo = async (
  platformName = 'instagram',
): Promise<ResponseData[]> => {
  const data = await axios
    .get(API_BASE + `/${platformName.toLowerCase()}`)
    .then((res) => {
      return res.data;
    });

  return data;
};
