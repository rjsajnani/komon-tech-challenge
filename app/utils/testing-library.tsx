import { render } from '@testing-library/react-native';

import { NativeBaseProvider } from 'native-base';
import { AppProvider } from '../context';

const inset = {
  frame: { x: 0, y: 0, width: 0, height: 0 },
  insets: { top: 0, left: 0, right: 0, bottom: 0 },
};

const renderWithProviders = (children: JSX.Element) => {
  return render(
    <AppProvider>
      <NativeBaseProvider initialWindowMetrics={inset}>
        {children}
      </NativeBaseProvider>
    </AppProvider>,
  );
};
export * from '@testing-library/react-native';

export { renderWithProviders };
