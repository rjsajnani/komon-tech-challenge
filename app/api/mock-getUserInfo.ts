import { ResponseData } from './getUserInfo';

export const mokcGetUserInfo = async (
  platformName = 'instagram',
): Promise<ResponseData[]> => {
  return Promise.resolve([
    {
      avatar_url: 'dummyUrl',
      bio: 'Bio from the creator',
      followers: '124',
      id: 1,
      name: 'Testing',
      lastestPost: [],
    },
  ]);
};
