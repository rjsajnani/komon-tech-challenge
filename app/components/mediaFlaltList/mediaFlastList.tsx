import { Box, FlatList, Image } from 'native-base';
import { LatestPost } from '../../api/getUserInfo';
import { ResizeMode, Video } from 'expo-av';

import { MeidaFlatListProps as Props } from './mediaFlaltList.props';
import { StyleSheet } from 'react-native';

// This component is handling the media provided by various social media API

export const MediaFlatList: React.FC<Props> = (props) => {
  const { lastestPost } = props;
  return (
    <FlatList
      numColumns={lastestPost[0].type === 'img' ? 1 : 3}
      data={lastestPost}
      renderItem={({ item }) => {
        return (
          <Box
            m={2}
            maxW="100%"
            rounded="lg"
            overflow="hidden"
            borderColor="coolGray.200"
            borderWidth="1"
            _dark={{
              borderColor: 'coolGray.600',
              backgroundColor: 'gray.700',
            }}
          >
            {item.type === 'img' ? (
              <Image
                size="2xl"
                py={2}
                px={2}
                width={'100%'}
                alt="/"
                source={{
                  uri: item.postImage,
                }}
              />
            ) : (
              <Video
                style={styles.video}
                source={{
                  uri: item.postImage,
                }}
                shouldPlay={true}
                useNativeControls
                resizeMode={ResizeMode.STRETCH}
                isLooping
              />
            )}
          </Box>
        );
      }}
      keyExtractor={(item) => item.postId.toString()}
    />
  );
};

const styles = StyleSheet.create({
  video: {
    alignSelf: 'center',
    width: 120,
    height: 200,
  },
});
