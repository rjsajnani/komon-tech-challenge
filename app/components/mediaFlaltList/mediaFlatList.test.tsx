import React from 'react';
import { LatestPost } from '../../api/getUserInfo';

import { renderWithProviders, RenderAPI } from '../../utils/testing-library';
import { MediaFlatList } from './mediaFlastList';

const mockMediaImg: LatestPost[] = [
  {
    postId: 1,
    postImage: 'string',
    type: 'img',
  },
];
const mockMediaVideo: LatestPost[] = [
  {
    postId: 1,
    postImage: 'string',
    type: 'img',
  },
];

describe('<MediaFlatList />', () => {
  let tree: RenderAPI;

  const createInstance = (type: LatestPost[]) => {
    tree = renderWithProviders(<MediaFlatList lastestPost={type} />);
  };

  it('renders image  type correctly', () => {
    createInstance(mockMediaImg);
    expect(tree).toMatchSnapshot();
  });

  it('renders video type correctly', () => {
    createInstance(mockMediaVideo);
    expect(tree).toMatchSnapshot();
  });
});
