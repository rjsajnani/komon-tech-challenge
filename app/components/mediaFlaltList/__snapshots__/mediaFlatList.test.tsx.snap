// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MediaFlatList /> renders image  type correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RCTScrollView
    contentContainerStyle={
      [
        {},
        {
          "dataSet": {},
        },
        {},
      ]
    }
    data={
      [
        {
          "postId": 1,
          "postImage": "string",
          "type": "img",
        },
      ]
    }
    dataSet={{}}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    style={{}}
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          dataSet={{}}
          style={
            {
              "borderColor": "#e5e7eb",
              "borderRadius": 8,
              "borderWidth": 1,
              "marginBottom": 8,
              "marginLeft": 8,
              "marginRight": 8,
              "marginTop": 8,
              "maxWidth": "100%",
              "overflow": "hidden",
            }
          }
        >
          <Image
            accessibilityLabel="/"
            alt="/"
            dataSet={{}}
            onError={[Function]}
            source={
              {
                "uri": "string",
              }
            }
            style={
              {
                "height": 256,
                "maxWidth": "100%",
                "paddingBottom": 8,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 8,
                "width": "100%",
              }
            }
          />
        </View>
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaProvider>
`;

exports[`<MediaFlatList /> renders video type correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RCTScrollView
    contentContainerStyle={
      [
        {},
        {
          "dataSet": {},
        },
        {},
      ]
    }
    data={
      [
        {
          "postId": 1,
          "postImage": "string",
          "type": "img",
        },
      ]
    }
    dataSet={{}}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    style={{}}
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          dataSet={{}}
          style={
            {
              "borderColor": "#e5e7eb",
              "borderRadius": 8,
              "borderWidth": 1,
              "marginBottom": 8,
              "marginLeft": 8,
              "marginRight": 8,
              "marginTop": 8,
              "maxWidth": "100%",
              "overflow": "hidden",
            }
          }
        >
          <Image
            accessibilityLabel="/"
            alt="/"
            dataSet={{}}
            onError={[Function]}
            source={
              {
                "uri": "string",
              }
            }
            style={
              {
                "height": 256,
                "maxWidth": "100%",
                "paddingBottom": 8,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 8,
                "width": "100%",
              }
            }
          />
        </View>
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaProvider>
`;
